web3_proxy/Cargo.toml

```diff
@@ -46,10 +46,12 @@ gethostname = "0.4.2"
 glob = "0.3.1"
 handlebars = "4.3.6"
 hashbrown = { version = "0.13.2", features = ["serde"] }
+hex_fmt = "0.3.0"
```

Comment:
What do we need hexfmt for? The ethers types should handle conversion to hex for us

Reply:
This is where I use it (in frontend/users/payments.rs). I was having troubles using the ethers types HexFmt(keccak256("get_approved_tokens()".to_owned().into_bytes())). I can fiddle around with it again. Let me know what you prefer (and if you happen to know a quick way to format this properly)

Reply:
Leave it for now and make a task. If it works, its fine for now. I made a quick test and this is much shorter and works for me. No need to go from &str -> String -> bytes. keccack256 treats str as utf-8 strings.

Both your code and this code give `0x79aaf078077c817b5d573b12ae04125ffe54cd32f0b7afeede391e27ee973ee9` (thought printing with json wraps in quotes since its a string and thats how JSON prints strings)

```
import ethers::prelude::H256;

let hash = H256::from(keccak256("get_approved_tokens()"));

println!("{:?}", hash);

println!("{}", serde_json::to_string(&hash).unwrap());

```

IIRC, you were trying to avoid using the "debug" formatter. But that is just how the ethers team decided to use it.